libs/collectives/docs/index.rst

```diff
@@ -0,0 +1,12 @@
+..
+   Copyright (c) 2019 The STE||AR-Group
+
+   Distributed under the Boost Software License, Version 1.0. (See accompanying
+   file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _libs_collectives:
+
+=====
+collectives
+=====
+
```

Comment:
Would you mind adding at least one sentence describing what the module provides? And this is absolutely not important, but I've tried to keep the over- and underlines of the heading the same length as the heading itself.